### ISSUE TITLE
Parameters to improve dense recon in textureless areas.

### ISSUE THREAD

**Issue by FishHe** [colmap/colmap#610]
COLMAP is awesome for outdoor scenes.
However, when I reconstruct the indoor scene, the dense-recon part of COLMAP is worse than OpenMVS and MVE. Is there any parameters to improve the behavior in textureless areas?
I notice that in COLMAP the smoothness term is not for depth but for view-selection. Would it be possible to add a depth-smoothness term like SGM?

**Comment by ahojnnes** [comment]
COLMAP is based on the patch match algorithm, as such it does not any explicit regularization for smoothness. Increasing the window_radius should help a little but will increase some other artifacts.

**Comment by FishHe** [comment]
Thanks for the advice~
Increasing the window_radius does help a little. Then, it takes really long time for each iteration.
I tried to increase the window_step to control time cost. The program brokes down, if window_step is set more than 2. 
The template function in patch_match_cuda.cu is just implemented for window_step of [1,2].
I added [3,4,5] for further experiment.
It makes some progress if window_radius and window_step is set properly.

**Comment by ahojnnes** [comment]
Indeed, you have to find a good combination of these parameters. Not all of them work well.

**Comment by FishHe** [comment]
I found papers to improve this, but the author will not open source in the near future.
Now I'm trying to repeat the ACMH and ACMM algorithms:
https://arxiv.org/abs/1805.07920
https://arxiv.org/abs/1904.08103v1
TAPA-MVS are also designed for this problem, but the idea is full of artificial traces:
https://arxiv.org/abs/1903.10929v1

I think it will be a great progress if COLMAP improves the behavior in low-tex areas.

**Comment by ahojnnes** [comment]
The ACMM algorithm seems to be the most promising approach from what I can tell after a quick glance. If you manage to implement this in CUDA integrated into COLMAP, I would be happy to merge this. I don't have time to implement this myself, but I'd be happy to help by iterating on it in a pull request.

**Comment by ahojnnes** [comment]
Closing as the maintainer explained that PatchMatch has no explicit smoothness regularization and suggested parameter tuning approaches.
